src/PooledArrays.jl

```diff
@@ -179,8 +179,7 @@ PooledArray
         throw(ArgumentError("Cannot construct a PooledArray with type $R with a pool of size $(length(pool))"))
     end
 
-    # Assertions are needed since _label is not type stable
-    return PooledArray(RefArray(refs::Vector{R}), invpool::Dict{T,R}, pool)
+    return PooledArray(RefArray(refs), invpool, pool)
```

Comment:
```suggestion
    # Assertions are needed since _label is not type stable
    return PooledArray(RefArray(refs::Array{ndims(d), R}), invpool::Dict{T,R}, pool)
```